CDCgov/prime-reportstream
# 6139 FHIRtoHL7  Duplicate Element Name Protection

This PR prevents multiple elements in a schema from having the same name.

Test Steps:
1. Run the converter with the `fhirdata` command. e.g. `./prime fhirdata --input-file src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir -s metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml --output-format=HL7`.
2. Verify the output is valid HL7.
3. Open any FHIRtoHL7 config (i.e. `patient-visit.yml`) and change the `name` value on multiple elements to be the same.
4. Repeat step 1, but now an error will appear pointing at the culprit.

## Changes
- Added a safety check for duplicate element names with an exception that shows the troublemakers.
- Added a new test case to cover this.

## Linked Issues
- #6139
